src/__tests__/Container.spec.ts

```diff
@@ -305,6 +305,29 @@ describe("Container", () => {
       let childContainerWithOverride = parentContainer.providesValue("value", 2);
       expect(childContainerWithOverride.get("service")).toBe(1);
     });
+
+    test("overriding with a different type changes resulting container's type", () => {
+      const parentContainer = Container.providesValue("value", 1);
+      let childContainerWithOverride = parentContainer.providesValue("value", "two");
+
+      // @ts-expect-error should be failing to compile as the type of the container has changed
+      let numberValue: number = childContainerWithOverride.get("value");
+
+      let value: string = childContainerWithOverride.get("value");
+      expect(value).toBe("two");
+
+      const partialContainer = new PartialContainer({
+        value: Injectable("value", () => "three"),
+      });
```

Comment:
nit: this syntax is a unusual to me
```suggestion
      const partialContainer = new PartialContainer({}).provides(Injectable("value", () => "three"));
```

Reply:
Updated, thanks!